src/settings/BatchGroupConfigurationSettings/BatchGroupsField.js

```diff
@@ -0,0 +1,47 @@
+import React, { useMemo } from 'react';
+import { FormattedMessage } from 'react-intl';
+import PropTypes from 'prop-types';
+
+import {
+  KeyValue,
+  Select,
+} from '@folio/stripes/components';
+
+const BatchGroupsField = ({
+  batchGroups,
+  setSelectedBatchGroupId,
+}) => {
+  const batchGroupsOptions = useMemo(() => {
+    if (!batchGroups) return [];
+
+    return batchGroups.map(({ name, id }) => ({ label: name, value: id })) || [];
+  }, [batchGroups]);
+
+  return batchGroups.length > 1
+    ? (
+      <Select
+        dataOptions={batchGroupsOptions}
+        fullWidth
+        label={<FormattedMessage id="ui-invoice.settings.batchGroupConfiguration.batchGroup" />}
+        onChange={({ target }) => setSelectedBatchGroupId(target.value)}
+        required
+      />
+    )
+    : (
+      <KeyValue
+        label={<FormattedMessage id="ui-invoice.settings.batchGroupConfiguration.batchGroup" />}
+        value={batchGroups[0]?.name}
+      />
+    );
+};
+
+BatchGroupsField.propTypes = {
+  batchGroups: PropTypes.arrayOf(PropTypes.object),
+  setSelectedBatchGroupId: PropTypes.func.isRequired,
+};
+
+BatchGroupsField.defaultProps = {
+  batchGroups: [],
```

Comment:
@folio-org/acquisitions-ui guys, I believe we need a discussion here. IMHO, we should have a rule  that  empty arrays shouldn't be default values anywhere. First reason - it hides the real value: if value was not assigned yet we can't show a spinner in a place of such component. Even if that is not a requirement for now - it's better to have this feature enabled.
Secondly, empty array doesn't equal to empty array in JS, so it's kind of performance reason. Basically, React will treat this prop as changed on every render.
Any thoughts?

Reply:
Agreed. Removed default props `batchGroups`

Reply:
1. it depends on case and requirements, but I agree it's not required here
2. I don't see any issues with array default props as defaultProps is static and it's the same array